src/openLibrary/producer/parser.js

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (C) 2018  Shivam Tripathi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+import {entityTypes, isNotDefined, sortName} from '../../helpers/utils';
+import {identifiers, mapLanguage} from '../../helpers/mapping';
+import _ from 'lodash';
+import franc from 'franc-min';
+
+
+const WORK = 'work';
+const EDITION = 'edition';
+const AUTHOR = 'author';
+
+
+function detectLanguage(name) {
+	let lang = franc(name);
+	lang = lang !== 'und' ? lang : 'eng';
+
+	return mapLanguage(lang);
+}
+
+function processWork(json) {
+	if (isNotDefined(json)) {
+		return null;
+	}
+
+	const work = {};
+	work.alias = [];
+	work.metadata = {};
+	work.metadata.relationships = [];
+	work.metadata.links = [];
+	work.identifiers = [];
+
+	work.entityType = entityTypes.WORK;
+	work.source = 'OPENLIBRARY';
+
+	// Set up aliases
+	// The first alias (in the logical flow) is made to be default alias
+	let defaultAlias = true;
+	if (!isNotDefined(json.title)) {
+		const name = json.title;
+		const lang = detectLanguage(name);
+
+		work.alias.push({
+			default: defaultAlias,
+			languageId: lang,
+			name,
+			primary: true,
+			sortName: sortName(name)
+		});
+		defaultAlias = false;
+	}
+	if (!isNotDefined(json.subtitle)) {
```

Comment:
I wasn't sure where to put it otherwise. Data I sampled mostly looked like it contained different/same string as the name - so it put it here. 
We can add a colon (or skip it if it already has a `?` or a `-`) to the name and store it as a separate alias?